meteor/sofie-server/data-access/repositories/mongo/mongo-segment-repository.ts

```diff
@@ -33,4 +34,22 @@ export class MongoSegmentRepository extends BaseMongoRepository implements Segme
 			})
 		)
 	}
+
+	public async deleteSegments(rundownId: string): Promise<void> {
+		this.assertDatabaseConnection(this.deleteSegments.name)
+		const segments = await this.getSegments(rundownId)
+
+		segments.map((segment) => segment.id).forEach(async (id) => this.partRepository.deleteParts(id))
+		const segmentDeleteResult = await this.getCollection().deleteMany({ rundownId: rundownId })
```

Comment:
This comment goes for all the Repositories:
Since we want to delete the "child" entities before the "parent" entities (as you are doing here) then this would be a good thing to write a test case about :)

Reply:
The requested tests should now exist.